client/webserver/site/src/js/markets.js

```diff
@@ -349,14 +381,22 @@ export default class MarketsPage extends BasePage {
     const page = this.page
     const market = this.market
     const asset = app.assets[assetID]
-    switch (assetID) {
-      case (market.base.id):
-        this.setBalance(asset, page.baseBalSpan, page.baseImg, page.baseNewButton, page.baseBalance)
-        this.walletIcons.base.readWallet(asset.wallet)
-        break
-      case (market.quote.id):
-        this.setBalance(asset, page.quoteBalSpan, page.quoteImg, page.quoteNewButton, page.quoteBalance)
-        this.walletIcons.quote.readWallet(asset.wallet)
+    const [b, q] = [market.base, market.quote]
+    if (!b) {
+      Doc.hide(page.baseBox)
+    } else if (b.id === assetID) {
+      Doc.show(page.baseBox)
+      this.setBalance(asset, page.baseBalSpan, page.baseImg, page.baseNewButton, page.baseBalance)
+      this.walletIcons.base.readWallet(asset.wallet)
+      return
+    }
+
+    if (!q) {
+      Doc.hide(page.quoteBox)
+    } else if (q.id === assetID) {
+      this.setBalance(asset, page.quoteBalSpan, page.quoteImg, page.quoteNewButton, page.quoteBalance)
+      this.walletIcons.quote.readWallet(asset.wallet)
+      Doc.show(page.quoteBox)
     }
```

Comment:
You're still potentially performing actions on both wallets instead of just the specified asset's wallet. `updateWallet` should only be checking the wallet for the specified asset. Return the `switch` block so that you're only performing actions on one wallet box. You will probably have to check `case (market.baseCfg.id):` instead of `case (market.base.id):` (and the same for the quote asset), but still set wallet visibility based on `b` or `q`.

```
switch (assetID) {
  case (market.baseCfg.id):
    if (!b) {
        Doc.hide(page.baseBox)
        return
    }
    Doc.show(page.baseBox)
    this.setBalance(asset, page.baseBalSpan, page.baseImg, page.baseNewButton, page.baseBalance)
    this.walletIcons.base.readWallet(asset.wallet)
    break
  case (market.quoteCfg.id):
    ...
```